src/runtime/server/middleware/session/index.ts

```diff
@@ -121,6 +121,19 @@ const getSession = async (event: H3Event): Promise<null | Session> => {
   return session
 }
 
+const getImmutableSession = (session: Session) => {
+  const immutableSession = { ...session }
+  const properties = ['id', 'createdAt', 'ip']
+
+  properties.forEach((property) => {
+    Object.defineProperty(immutableSession, property, {
+      writable: false
+    })
+  })
+
+  return immutableSession as Session
```

Comment:
can you update the `Session` interface so that the three properties are also immutable?